src/core/error.rs

```diff
@@ -85,8 +97,8 @@ impl ContractError {
         Err(self)
     }
     /// A simple abstraction to wrap an error response just by passing the message
-    pub fn std_err<T>(msg: impl Into<String>) -> Result<T, ContractError> {
-        Err(ContractError::Std(StdError::generic_err(msg)))
+    pub fn std_err(msg: impl Into<String>) -> ContractError {
```

Comment:
This function was named weird for a thing that dynamically returned a `Result`, so I just changed it to return the contract error itself and had the call sites chain into `to_result` like they should be doing anyway